Spigot-API-Patches/0221-Create-HoverEvent-from-ItemStack-Entity.patch

```diff
@@ -0,0 +1,108 @@
++        return server.hoverContentOf(entity, customName);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable BaseComponent customName) {
++        return server.hoverContentOf(entity, customName);
++    }
+     // Paper end
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 7c0a788900c93c29d14d8c45ac5ae3317cf4a94e..ad1a81817aabaa199227cfa8da4af4232cbb75ab 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -13,6 +13,8 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++import net.md_5.bungee.api.chat.BaseComponent; // Paper
++import net.md_5.bungee.api.chat.hover.content.Content; // Paper
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1555,5 +1557,34 @@ public interface Server extends PluginMessageRecipient {
+      */
+     @NotNull
+     com.destroystokyo.paper.entity.ai.MobGoals getMobGoals();
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack for displaying.
++     *
++     * @param itemStack
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack
++     */
++    @NotNull
++    Content hoverContentOf(@NotNull ItemStack itemStack);
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    Content hoverContentOf(@NotNull Entity entity, @Nullable String customName);
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    Content hoverContentOf(@NotNull Entity entity, @Nullable BaseComponent customName);
```

Comment:
Should this not be `BaseComponent[]`?

Reply:
We have only one TextComponent that is one BaseComponent.